src/async_hta_service.hpp

```diff
@@ -442,6 +450,89 @@ class AsyncHtaService
         return it.first->second;
     }
 
+private:
+    class Stats
+    {
+    public:
+        template <typename T>
+        void add_read_duration(T duration)
+        {
+            {
+                std::scoped_lock lock(stats_lock_);
+                read_count_++;
+                read_duration_ +=
+                    std::chrono::duration_cast<std::chrono::duration<double>>(duration).count();
+            }
+            log_stats_();
+        }
+
+        template <typename T>
+        void add_write_duration(T duration)
+        {
+            {
+                std::scoped_lock lock(stats_lock_);
+                write_count_++;
+                write_duration_ +=
+                    std::chrono::duration_cast<std::chrono::duration<double>>(duration).count();
+            }
+            log_stats_();
+        }
+
+        void increment_ongoing()
+        {
+            std::scoped_lock lock(stats_lock_);
+            ongoing_requests_count_++;
+            log_stats_();
+        }
+
+        void decrement_ongoing()
+        {
+            std::scoped_lock lock(stats_lock_);
```

Comment:
Remove the lock and call this function only in the functions `add_write_duration` and `add_read_duration`